File: b_rank_new_level_up_problems/b_rank_new_level_up_problems__perfect_shuffle.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    class Cards
    {
        public string Card { get; set; }
        public int Num { get; set; } = 0;

        public Cards( string s )
        {
            Card = s;
        }
    }

    static void Main()
    {
        int k = int.Parse( ReadLine() );
        string[] kind = { "S", "H", "D", "C" };
        var l = new System.Collections.Generic.List<Cards>( 52 );

        SetOut( new System.IO.StreamWriter( OpenStandardOutput() ) { AutoFlush = false } );
        foreach( int i in Range( 0, 4 ) ) {
            foreach( int j in Range( 1, 13 ) ) {
                l.Add( new Cards( $"{kind[i]}_{j}" ) );
            }
        }
        foreach( int _ in Range( 0, k ) ) {
            foreach( int i in Range( 0, l.Count ) ) {
                l[i].Num = i < 26 ? i * 2 : ( i - 26 ) * 2 + 1;
            }
            l.Sort( ( a, b ) => a.Num - b.Num );
        }
        foreach( Cards c in l ) {
            WriteLine( c.Card );
        }
        Out.Flush();
        return;
    }
}

File: b_rank_new_level_up_problems/b_rank_new_level_up_problems__perfect_shuffle.py
# 実行時間 20ms
k = int(input())
kd = ["S", "H", "D", "C"]
l = []

for i in range(4):
    for j in range(1, 14):
        l.append([f"{kd[i]}_{j}", 0])
for _ in range(k):
    for i in range(len(l)):
        l[i][1] = i * 2 if i < 26 else (i - 26) * 2 + 1
    l.sort(key=lambda x: x[1])
for i in l:
    print(i[0])
